internal/datasources/rest/handler.go

```diff
@@ -0,0 +1,207 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/santhosh-tekuri/jsonschema/v6"
+	uritemplate "github.com/std-uritemplate/std-uritemplate/go/v2"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/mindersec/minder/internal/util/schemaupdate"
+	"github.com/mindersec/minder/internal/util/schemavalidate"
+	minderv1 "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+const (
+	// MaxBytesLimit is the maximum number of bytes to read from the response body
+	// We limit to 1MB to prevent abuse
+	MaxBytesLimit int64 = 1 << 20
+)
+
+type restHandler struct {
+	// nis is a nullable input schema
```

Comment:
That's right. Though I'm starting to regret this and will.most likely enforce that it can't be null in a subsequent iteration. More constraints make this a little simpler.

Reply:
The comment was confusing... just calling it `schema` and `rawSchema` with a comment that it could be null felt clearer to me.